crates/rpc-client/src/rpc_client.rs

```diff
@@ -996,15 +1002,36 @@ impl RPCClient {
         last_finalized_block_number: u64,
         min_capacity: Option<u64>,
         max_cells: usize,
+    ) -> Result<QueryResult<CollectedCustodianCells>> {
+        Self::query_finalized_custodian_cells_from_indexer(
+            &self.indexer,
+            &self.rollup_context,
+            withdrawals_amount,
+            custodian_change_capacity,
+            last_finalized_block_number,
+            min_capacity,
+            max_cells,
+        )
+        .await
+    }
+
+    // TODO: refactor into custodian mod
+    #[instrument(skip_all)]
+    async fn query_finalized_custodian_cells_from_indexer(
+        indexed_cells: &impl IndexedCells,
```

Comment:
It's a bit strange having a testing injection point here. Could it be pushed to lower layers, e.g. in `request`?

Reply:
It is feasible to add some `cfg(test)` only code to `CKBIndexerClient` for mocking responses, e.g. `set_mock_response(response_callback)`. This way you won't need the `IndexedCells` trait and this `query_finalized_custodian_cells_from_indexer` method, and other RPC methods can be more easily tested too.

Reply:
I think standalone mock struct for test only is more flexible. I plan to refactor all custodian query code into custodian mod. It doesn't need to care about how to construct `CKBIndexerClient`.

Then in `rpc_client`, we simply call ```custodian::query_finalized_cells``` instead.

Reply:
OK then.